### ISSUE TITLE
Need each version outputs server and client

### ISSUE THREAD

**Issue by nak3** [openshift/origin#5563]
Issue
---
`openshift version` should output server version. Current version is client side only.

- Current version output
~~~
$ openshift version
openshift v1.0.6-803-g7c12c7a-dirty
kubernetes v1.2.0-alpha.1-1107-g4c8e6f4
etcd 2.1.2
~~~

- TOBE
~~~
$ openshift version
Client:
 openshift v1.0.6-803-g7c12c7a-dirty
 API v1

Server:
 openshift v1.0.6-803-g7c12c7a-dirty
 kubernetes v1.2.0-alpha.1-1107-g4c8e6f4
 etcd 2.1.2
 API v1
~~~

**Comment by liggitt** [comment]
`openshift version` does output the server version

`oc version` ouputs the client version

**Comment by nak3** [comment]
I mean that CLI(both `openshift version` and `oc version`) should get server version by REST API.
Since how does clients know server's support API version?
I know it outputs in its error message, but we need appropriate method.

~~~
$ oc create -f tmp.json
error: could not read an encoded object: API version "v1beta1" in "hello-pod.json" isn't supported, only supports API versions ["v1" "v1beta3"]
~~~

**Comment by liggitt** [comment]
`oc version` is the version of the executable, and should not make any server calls

**Comment by nak3** [comment]
So, there are no way to know openshift's exposed API version from clients? (except for the error message above.)

**Comment by deads2k** [comment]
> So, there are no way to know openshift's exposed API version from clients? (except for the error message above.)

That information would be exposed by `oc api-versions` (mirror of a kube command), but the kube command is broken and depends on a concerning number of cherry-picks both server and client side.  I don't think we're likely to enable it for 3.1.

**Comment by nak3** [comment]
I beleve there are several problem in current version outputs.

- 1. Need several steps to collect version

If we hit some bugs and want to check the versions, we need some steps:
1. Run `oc version` -> 2. ssh master server -> 3. run `openshift version`
In case that we don't have permission to access to the master server, we will not be able to know server versions.

And if we run `openshift version` on another host, we may misunderstand the version. We need care about it.

- 2. Output embeded etcd version

Since `openshift version` outputs embeded etcd version, the output would be wrong if they use external etcd.

- 3. Can't get API version.

`oc api-versions` doesn't work at this moment.

**Comment by deads2k** [comment]
fixed in 3.3
